homeassistant/components/smarttub/controller.py

```diff
@@ -78,13 +78,20 @@ async def async_update_data(self):
         data = {}
         try:
             async with async_timeout.timeout(POLLING_TIMEOUT):
-                for spa in self.spas:
-                    data[spa.id] = {"status": await spa.get_status()}
+                # Dict[spa_id, Dict]
+                data = dict(
+                    await asyncio.gather(
```

Comment:
If you want to keep it sequential to avoid rate limits that's a good reason. This optimization is optional and not all devices or services work well with it.

Reply:
OK, I'll do that. I can't test the case of multiple spas in an account, so I'm more comfortable being conservative